centreon/www/install/php/Update-24.04.8.php

```diff
@@ -80,19 +98,24 @@
 
     $pearDB->commit();
 } catch (\Exception $e) {
+    CentreonLog::create()->error(
+        logTypeId: CentreonLog::TYPE_UPGRADE,
+        message: $versionOfTheUpgrade . $errorMessage,
+        customContext: ['trace' => $e->getTraceAsString()],
+        exception: $e
+    );
 
     if ($pearDB->inTransaction()) {
-        $pearDB->rollBack();
+        try {
+            $pearDB->rollBack();
+        } catch (PDOException $e) {
+            CentreonLog::create()->error(
+                logTypeId: CentreonLog::TYPE_UPGRADE,
+                message: "{$versionOfTheUpgrade} error while rolling back the upgrade operation",
+                exception: $e
+            );
+        }
```

Comment:
This is a good reflex :)

What follows is a highlight, not a request change.

Could we consider providing a way to make this part generic and common to every migrations ?
I was going to suggest to put it in the parent class, but it's not a class. 

In any case it's relevant for every migrations, so we are going to face redundancy if we don't think about how to factorize it.

Reply:
Thanks :) We can create a class dedicated at the upgrade with two sections : DDL statements and DML statements.